### PR TITLE
Fix to CA2119 and #1086: Ignore classes that cannot be subtyped outside assembly

### PR DESCRIPTION
**Pull request by alejandro5042**
By definition, this rule should not fire if the class cannot be subclassed outside the current assembly. Classes that are declared as 'internal' fit the bill, as well as classes that do not have public constructors. This change addresses these gaps.

### CODE REVIEW

File: src/Microsoft.QualityGuidelines.Analyzers/Core/SealMethodsThatSatisfyPrivateInterfaces.cs [hunk 1]
```diff
@@ -39,13 +41,15 @@ public override void Initialize(AnalysisContext analysisContext)
...
-
+        
         private static void CheckTypes(SymbolAnalysisContext context)
         {
-            var type = (ITypeSymbol)context.Symbol;
+            var type = (INamedTypeSymbol)context.Symbol;
 
-            // only classes can have overridable members
-            if (type.TypeKind == TypeKind.Class)
+            // Only classes can have overridable members, and furthermore, only consider classes that can be subclassed outside this assembly.
```
Comment (mavasani) [inline src/Microsoft.QualityGuidelines.Analyzers/Core/SealMethodsThatSatisfyPrivateInterfaces.cs]: Please add a comment that internal types that can be instantiated from either this assembly or other assemblies which have InternalsVisibleTo access. For both these cases, there should be no security vulnerabilities introduced by overriding methods, hence these types can be ignored.

Comment (alejandro5042) [inline src/Microsoft.QualityGuidelines.Analyzers/Core/SealMethodsThatSatisfyPrivateInterfaces.cs]: Good point. Done!

**Comment by alejandro5042** [comment]
I believe we also need to check if the class is sealed. If you merge this PR before I make this change, I'll begin another.

**Comment by mavasani** [comment]
Thanks, I'll wait for your commit.

**Pull request merged**